### PR TITLE
GraphQL Support for withdrawals

### PR DESCRIPTION
**Pull request by shemnon**
Add support for withdrawals in GraphQL, including needed changes to testing infrastructure for shanghai-era blocks.
Also align existing adapters with graphql schema optionality.

### CODE REVIEW

**Comment by shemnon** [comment]
partial work to address https://github.com/ethereum/execution-apis/pull/400

File: ethereum/api/src/test/resources/org/hyperledger/besu/ethereum/api/graphql/eth_getBlock_shanghai.json
[the remark this answers is not in the document]
Comment (shemnon) [reply ethereum/api/src/test/resources/org/hyperledger/besu/ethereum/api/graphql/eth_getBlock_shanghai.json:1]: I have a followup for hex encoding that I'll post today.

For the record, the EIP was integer encoded, as were the initial hive tests.  The execution tests transition to hex encoding was never subject to public discussion.

[the remark this answers is not in the document]
Comment (shemnon) [reply ethereum/api/src/test/resources/org/hyperledger/besu/ethereum/api/graphql/eth_getBlock_shanghai.json:1]: Migration to strings - https://github.com/hyperledger/besu/pull/5505 - Note that it builds off of this PR.  It is only really a change to Scalars.java and most of the unit test vectors.

**Review by gfukushima — APPROVED** [review]
lgtm

File: ethereum/api/src/test/resources/org/hyperledger/besu/ethereum/api/graphql/eth_getBlock_shanghai.json
[the remark this answers is not in the document]
Comment (s1na) [reply ethereum/api/src/test/resources/org/hyperledger/besu/ethereum/api/graphql/eth_getBlock_shanghai.json:1]: > For the record, the EIP was integer encoded, as were the initial hive tests.

The first geth implementation by Nick Johnson used hex-encoding. At some point half of the fields were changed to integer encoding (including hive tests for them) and half were kept as hex which was weird. I proposed on the specs repo to change it to hex.

> The execution tests transition to hex encoding was never subject to public discussion.
I took this comment https://github.com/ethereum/execution-apis/pull/389#issuecomment-1480463374 as an approval from the Besu side. I'm sorry if you thought there wasn't enough discussion around it. I'm planning some more changes to the hive tests, how do you prefer to discuss them?

**Pull request merged**